app/services/API.js

```diff
@@ -37,18 +39,20 @@ const API = {
         });
     },
 
-    sync: (revision, accessToken) => {
+    sync: (revision, validator) => {
         console.tron.log("sync with rev " + revision);
         return new Promise((resolve, reject) => {
-            executeRequest('POST', 'sync', { revision: revision }, accessToken)
-            .then((json) => {
-                // TODO: see if need undefined checks
-                console.tron.log("sync success " + json);
-                resolve(json);
-            }).catch((error) => {
-                // TODO: check if this will work, I'm assuming that the error will always be an action
-                reject(error);
-            });
+            if (validator.isValid) {
```

Comment:
If it's invalid, then you never fulfill the promise, which means you have sagas just hanging there.

Have it be like... `if (!validator.isValid) { reject(error); }` except create the actual error.